search-app/src/cmr/search/api/association.clj

```diff
@@ -84,6 +96,30 @@
   (validate-association-content-type headers)
   (info (format "Dissociating %s [%s] from collections: %s by client: %s."
                 (name concept-type) concept-id body (:client-id context)))
-  (let [results (assoc-service/dissociate-from-collections context concept-type concept-id body)
-        status-code (association-results->status-code concept-type results)]
-    (api-response status-code results)))
+  (if-not (valid-association-args-count? concept-type
+                                         (map :concept-id (json/parse-string body true))) 
+    (api-response
+      400
+      {:error "Only one variable at a time may be dissociated."})
+    (let [results (assoc-service/dissociate-from-collections context concept-type concept-id body)
+          status-code (association-results->status-code concept-type results)]
+      (api-response status-code results))))
+
+(def association-api-routes
+  (context "/associations" []
+    (context "/variables" []             
+      (context "/:variable-concept-id" [variable-concept-id]
+        (context "/collections" []
+          ;; Associate variable with a collection
+          (POST "/:collection-concept-id" [collection-concept-id
+                                           req :as {:keys [request-context headers]}]
+                ;; TODO get rid of redundant json->string conversion
+                (associate-concept-to-collections
+                  request-context headers (json/generate-string [{:concept-id collection-concept-id}]) :variable variable-concept-id))
+          
+          ;; Dissociate a variable from a collection
+          (DELETE "/:collection-concept-id" [collection-concept-id
+                                             req :as {:keys [request-context headers]}]
+                  ;; TODO get rid of redundant json->string conversion
+                  (dissociate-concept-from-collections
+                    request-context headers (json/generate-string [{:concept-id collection-concept-id}]) :variable variable-concept-id)))))))
```

Comment:
@ygliuvt , you guys don't want to do end point like `%CMR-ENDPOINT%/associate/variables/V1200000008-PROV1/collections/C1200000005-PROV1`

Reply:
Personally, I would like to see all the association routes in one place, rather than some here and some in the cmr.search.api.variables.

Reply:
@sxu123 I feel like consolidating them would make sense. But right now only variable has the limit on association numbers. The others could be moved here eventually and deprecate the others routes. That can be discussed further.

Reply:
You don't need to consolidate others, just put all the variable associations together in the variable.clj.

Reply:
It's no big deal. If you want to do it this way, then mark which one is preferred way and which will be deprecated.